### PR TITLE
[fix](alter) fix potential concurrent issue for alter when check olap table state normal outside write lock scope is not atomic

### PR DESCRIPTION
**Pull request by caiconghui**
## Proposed changes

Issue Number: close #xxx

now, we check some olap table state normal outside write lock scope, the table state may be changed to unnormal when we do alter operation

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by caiconghui** [comment]
run buildall

**Comment by caiconghui** [comment]
run buildall

**Comment by caiconghui** [comment]
run buildall

**Comment by caiconghui** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -2471,13 +2466,6 @@ public void modifyTableLightSchemaChange(Database db, OlapTable olapTable,
             throws DdlException {
 
         LOG.debug("indexSchemaMap:{}, indexes:{}", indexSchemaMap, indexes);
-        if (olapTable.getState() == OlapTableState.ROLLUP) {
-            throw new DdlException("Table[" + olapTable.getName() + "]'s is doing ROLLUP job");
-        }
-
-        // for now table's state can only be NORMAL
-        Preconditions.checkState(olapTable.getState() == OlapTableState.NORMAL, olapTable.getState().name());
-
```
Comment (Tanya-W) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:2480]: Is ok removing this check here? I think here also need `olapTable.checkNormalStateForAlter();`

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 2]
```diff
@@ -1489,21 +1489,9 @@ public long getReplicaCount() {
...
     }
 
-    public void checkStableAndNormal(String clusterName) throws DdlException {
+    public void checkNormalStateForAlter() throws DdlException {
         if (state != OlapTableState.NORMAL) {
-            throw new DdlException("Table[" + name + "]'s state is not NORMAL. "
-                    + "Do not allow doing materialized view");
-        }
-        // check if all tablets are healthy, and no tablet is in tablet scheduler
-        boolean isStable = isStable(Env.getCurrentSystemInfo(),
```
Comment (Tanya-W) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:1498]: why remove this logic which check all tablets healthy?

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java (hunk 1 above)
Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:2480]: we already check table state inside the table write lock no need double check here

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java (hunk 2 above)
Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:1498]: actually checkTableStable function will check table stable, no need to double check here

**Review by weizuo93 — APPROVED** [review]
LGTM

**Pull request merged**